packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js

```diff
@@ -58,6 +67,10 @@ function parseInfo(context, info) {
       api.title = info.get('title');
       api.attributes.set('version', info.get('version'));
 
+      if (info.get('termsOfService')) {
+        api.links.push(parseTermsOfService(info.get('termsOfService')));
```

Comment:
The pattern in the code-base is generally in this block of code the everything is already parsed. I'd move the `parseTermsOfService` up a bit for consistency into something like:

```js
  const parseTermsOfService = pipeParseResult(namespace,
    parseString(context, name, false),
    getValue,
    (termsOfService) => {
      const link = new namespace.elements.Link();
      link.relation = 'terms-of-service';
      link.href = termsOfService;
      return link;
    });
```

Then parse the member as: `[hasKey('termsOfService'), parseTermsOfService],` then here it's just about connecting it up:

```js
      if (info.get('termsOfService')) {
        api.links.push(info.get('termsOfService'));
      }
```

Reply:
refactor on next commit